server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/JdbcDataTypeUtil.java

```diff
@@ -269,7 +269,8 @@ public static JDBCType parseDataType(String dataType) {
                 || "SIGNTYPE".equalsIgnoreCase(dataType)
                 || "SIMPLE_INTEGER".equalsIgnoreCase(dataType)
                 || "MEDIUMINT".equalsIgnoreCase(dataType)
-                || "INTEGER".equalsIgnoreCase(dataType)) {
+                || "INTEGER".equalsIgnoreCase(dataType)
+                || "YEAR".equalsIgnoreCase(dataType)) {
             dataType = "INTEGER";
```

Comment:
1. we'd better distinguish parseDataType with db type. cause same type name in different db may represent different JDBCType.
2. why not fist check if we have matched the type, if not match then do if else logic match

Reply:
That is the problem of the jdbc layer, I do that is just an opportunistic practice. I will communicate with people   responsible for jdbc to decide  the specific practice

Reply:
![image](https://github.com/user-attachments/assets/ea31d79e-e3f5-4849-b655-3ca33432f0bc)
have been parsed to `SMALLINT `,consistent with the jdbc specification